std/python/Boot.hx

```diff
@@ -276,129 +276,107 @@ class Boot {
 		return UBuiltins.isinstance(o, UBuiltins.list);
 	}
 
+
+
 	static function simpleField( o : Dynamic, field : String ) : Dynamic {
 		if (field == null) return null;
 
 		var field = handleKeywords(field);
 		return if (UBuiltins.hasattr(o, field)) UBuiltins.getattr(o, field) else null;
 	}
 
-	static var closures:Dynamic;
+	@:ifFeature("closure_Array", "closure_String")
+	static inline function createClosure (obj:Dynamic, func:Dynamic):Dynamic {
+		return new MethodClosure(obj, func);
+	}
 
-	static function getClosure (funId:String, o:Dynamic,  or : Void->Dynamic) {
-		
-		if (closures == null) {
-			python.Syntax.pythonCode("import weakref");
-			closures = python.Syntax.pythonCode("weakref.WeakValueDictionary()");
+	static function field( o : Dynamic, field : String ) : Dynamic {
+		if (field == null) return null;
+
+		if (isString(o)) {
+			return stringClosure(field, o);
 		}
-		var hash = UBuiltins.str(UBuiltins.id(o)) + "_" + funId;
-		var closure = python.Syntax.callField(closures, "get", hash);
-		if (closure == null) {
-			var res = or();
-			python.Syntax.arraySet(closures, hash, res);
-			return res;
+		if (isArray(o)) {
+			return arrayClosure(field, o);
 		}
-		return closure;
+
+		var field = handleKeywords(field);
+		return if (UBuiltins.hasattr(o, field)) UBuiltins.getattr(o, field) else null;
 	}
 
-	static function field( o : Dynamic, field : String ) : Dynamic {
+	static function stringClosure(  field : String, o : Dynamic ) : Dynamic {
+		switch (field) {
```

Comment:
This could probably be simplified quite a bit:

```haxe
return createClosure(o, switch (field) {
    case "toLowerCase": StringImpl.toLowerCase;
    case "toUpperCase": StringImpl.toUpperCase;
    // [...]
}
```